pkg/grpc/exec/exec.go

```diff
@@ -70,6 +72,36 @@ type MsgExecveEventUnix struct {
 	processapi.MsgExecveEventUnix
 }
 
+func (msg *MsgExecveEventUnix) RetryInternal(ev notify.Event, timestamp uint64) (*process.ProcessInternal, error) {
+	return nil, fmt.Errorf("Unreachable state: MsgExecveEventUnix with missing internal")
+}
+
+func (msg *MsgExecveEventUnix) Retry(internal *process.ProcessInternal, ev notify.Event) error {
+	var podInfo *tetragon.Pod
+
+	p := ev.GetProcess()
+	containerId := p.Docker
+	filename := p.Binary
+	args := p.Arguments
+	nspid := msg.Process.NSPID
+
+	if option.Config.EnableK8s && nspid > 0 {
```

Comment:
@tpapagian this is interesting, the tests are failing because I added this 'nspid > 0' clause. My logic was if we don't have a nspid then we are in the host pidns and so probably don't need to do a pod lookup. But on 2nd though if hostpid=true (is that even a thing) then we probably should still get a pod. So, I'll revert and replace it with 'contaienrId != 0' I think this should be more accurate. wdyt?

Reply:
I was trying to avoid bouncing through k8s lookups if it wasn't needed.

Reply:
Yes, this sounds reasonable. `contaienrId != ""` is also consistent with what we use inside `Needed()`.